### PR TITLE
feat: Add test infrastructure for findings postgres crate

### PR DESCRIPTION
**Pull request by ahrav**
Adds `test_postgres.rs` and `tests.rs` to the `gossip-findings-postgres` crate.

`test_postgres.rs` provides helpers for managing PostgreSQL test instances,
either via testcontainers or an external URL. It also includes logic for creating isolated databases and connections.

`tests.rs` contains live integration tests for schema and migration runner
functionality, covering table creation, index verification, foreign key constraints, and various data integrity checks.

## Summary by CodeRabbit

* **Tests**
  * Added test-only modules and integration test infrastructure for PostgreSQL with per-test isolated databases and lifecycle management (containerized or external).
  * Improved unique-name scheme (nonce/pid/counter) for test DBs and helpers to create per-test databases; options to apply or skip migrations.
  * Added modular, quote/escape-aware connection-string parsing and rewriting for both URI and keyword forms, with unit tests for edge cases.
  * Added comprehensive integration suite validating migrations, schema, constraints, relations, concurrency, and many edge cases.

### CODE REVIEW

File: crates/gossip-findings-postgres/src/tests.rs [hunk 1]
```diff
@@ -0,0 +1,1048 @@
...
+
+    thread_a
+        .join()
+        .expect("first migration thread should not panic")
+        .expect("first migration thread should succeed");
+    thread_b
+        .join()
+        .expect("second migration thread should not panic")
+        .expect("second migration thread should succeed");
+}
```
[the remark this answers is not in the document]
Comment (ahrav) [reply crates/gossip-findings-postgres/src/tests.rs]: Agreed — added a post-condition assertion after both threads join:

```rust
assert_eq!(
    row_count(&mut client, schema::SCHEMA_MIGRATIONS_TABLE),
    MIGRATIONS.len() as i64,
    "each migration must appear exactly once regardless of concurrent application"
);
```

This catches the scenario where one worker applies migrations while the other silently commits an empty transaction.

File: crates/gossip-findings-postgres/src/test_postgres.rs [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+    let authority_start = scheme_sep + 3;
+
+    match base_url[authority_start..].find('/') {
+        Some(path_offset) => {
+            let path_start = authority_start + path_offset;
+            format!("{}/{db_name}", &base_url[..path_start])
+        }
+        None => format!("{base_url}/{db_name}"),
+    }
+}
```
[the remark this answers is not in the document]
Comment (ahrav) [reply crates/gossip-findings-postgres/src/test_postgres.rs]: I looked at this and the function behaves correctly for its domain. PostgreSQL connection URIs use a single path segment for the database name — `postgresql://host:5432/dbname` is the standard format. Multi-segment paths like `postgresql://host:5432/proxy/main` are not valid PostgreSQL connection URIs.

The function's job is specifically to replace the database path component, and replacing everything from the first `/` after the authority is the correct behavior for that. Adding a `debug_assert` here would flag inputs that are malformed for this context, not legitimate edge cases being silently dropped. Leaving as-is.

File: crates/gossip-findings-postgres/src/tests.rs [hunk 3]
```diff
@@ -0,0 +1,1351 @@
...
+
+fn assert_has_expected_indexes(actual: &BTreeSet<String>, expected: &[&str]) {
+    for name in expected {
+        assert!(
+            actual.contains(*name),
+            "expected index {name:?} to exist; actual indexes: {actual:?}"
+        );
+    }
+}
```
[the remark this answers is not in the document]
Comment (ahrav) [reply crates/gossip-findings-postgres/src/tests.rs]: Applied. `table_index_names` → `table_indexes` now returns `BTreeMap<String, String>` mapping index name to `indexdef`, and `assert_has_expected_indexes` verifies each expected column name appears in the definition.

For example, `findings_tenant_secret_hash_idx` now asserts `["tenant_id", "secret_hash", "finding_id"]` all appear in the `indexdef` string. A migration that recreates an index on the wrong column list would be caught.
